### PR TITLE
Speed up and make more accurate relief mapping using binary search

### PR DESCRIPTION
**Pull request by RealBadAngel**
Picture taken with parallax scale 0.2 ;way beyond normal one to show accuracy

### CODE REVIEW

**Comment by RealBadAngel** [comment]
screenshot_20151209_071316

**Comment by nerzhul** [comment]
:+1:

**Comment by MirceaKitsune** [comment]
:+1:

File: client/shaders/nodes_shader/opengl_fragment.glsl [hunk 1]
```diff
@@ -47,15 +47,28 @@ vec4 get_normal_map(vec2 uv)
...
+	}	
+	depth = best_depth;
+	for (int i = 0; i < 4; i++) {
+		size *= 0.5;
+		float h = texture2D(normalTexture,dp + ds * depth).a;
+		if (depth <= h) {
+			best_depth = depth;
+			depth += 2.0 * size;
+		}
+		depth -= size;
```
Comment (ShadowNinja) [inline client/shaders/nodes_shader/opengl_fragment.glsl]: Why didn't you do this?
```GLSL
if (depth <= h) {
	best_depth = depth;
	depth += size;
} else {
	depth -= size;
}
```

Comment (RealBadAngel) [inline client/shaders/nodes_shader/opengl_fragment.glsl]: good point. will change it when i get back home

**Comment by RealBadAngel** [comment]
@ShadowNinja updated

**Comment by est31** [comment]
a64d78a37e95b99f32c29c8dec2e0cdae85a8b41

**Pull request closed without merging**